disco/models/upgrade_cost_analysis_generic_model.py

```diff
@@ -0,0 +1,360 @@
+        title="num_of_buses_with_voltage_violations",
+        description="Number of buses with voltage violations",
+    )
+    num_of_overvoltage_violations_buses_above_voltage_upper_limit: int = Field(
+        title="num_of_overvoltage_violations_buses_above_voltage_upper_limit",
+        description="Number of violations with buses above voltage_upper_limit",
+    )
+    voltage_upper_limit: float = Field(
+        title="voltage_upper_limit",
+        description="Voltage upper limit",
+        units="pu",
+    )
+    num_of_undervoltage_violations_buses_below_voltage_lower_limit: int = Field(
+        title="num_of_undervoltage_violations_buses_below_voltage_lower_limit",
+        description="Number of violations with buses below voltage_lower_limit",
+    )
+    voltage_lower_limit: float = Field(
+        title="voltage_lower_limit",
+        description="Voltage lower limit",
+        units="pu",
+    )
+    max_line_loading: float = Field(
+        title="max_line_loading",
+        description="Maximum line loading",
+        units="pu",
+    )
+    max_xfmr_loading: float = Field(
+        title="max_xfmr_loading",
+        description="Maximum transformer loading",
+        units="pu",
+    )
+    num_of_lines_with_violations_above_line_upper_limit: int = Field(
+        title="num_of_lines_with_violations_above_line_upper_limit",
+        description="Number of lines with violations above upper limit",
+    )
+    line_upper_limit: float = Field(
+        title="line_upper_limit",
+        description="Line upper limit",
+        units="pu",
+    )
+    num_of_xfmrs_with_violations_above_xfmr_upper_limit: int = Field(
+        title="num_of_xfmrs_with_violations_above_xfmr_upper_limit",
+        description="Number of transformers with violations above upper limit",
+    )
+    xfmr_upper_limit: float = Field(
+        title="xfmr_upper_limit",
+        description="xfmr upper limit",
+        units="pu",
+    )
+
+
+class EquipmentTypeUpgradeCostsModel(UpgradeParamsBaseModel):
+    """Provides costs for upgrading a type of equipment."""
+
+    name: str = Field(
+        title="name",
+        description="Job name",
+    )
+    type: str = Field(
+        title="type",
+        description="Equipment type",
+    )
+    count: str = Field(
+        title="count",
+        description="Count upgraded",
+    )
+    total_cost_usd: float = Field(
+        title="total_cost_usd",
+        description="Total cost in US dollars",
+        units="dollars",
+    )
+    # TODO: The comment field may be split into multiple fields.
+    comment: str = Field(
+        title="comment",
+        description="If the exact unit cost is not available for an equipment, unit cost for "
+            "the closest rated equipement will be considered for the upgrade cost computation. "
+            "This field will document which unit cost was considered.",
+    )
+
+
+class UpgradeSummaryResultsModel(UpgradeParamsBaseModel):
```

Comment:
FYI. Same here, the output metrics may be changed as well, we'll need to update the above models if happen.